src/client/components/terminal/index.js

```diff
@@ -3,13 +3,68 @@ import Markup from "preact-markup";
 
 import "./style.scss";
 
+import SearchButton from "../search-button";
+
 export default class Chart extends Component {
-  render(props) {
+  state = {
+    showActionBtn: false,
+    text: null,
+    googleBaseUrl: `https://www.google.com/search?q=`,
+    stackBaseUrl: `https://stackoverflow.com/search?q=`
+  };
+  // I don't want a re-render on mouse events
+  _highlightState = {
+    started: false,
+    mouseX: 0
+  };
+  componentDidMount() {
+    // mousedown on our tag === potential highlight
+    this.base.addEventListener("mousedown", this.onHighlightStart);
+    // a click anywhere === potential highlight end or highlight dismiss
+    window.addEventListener("mouseup", this.highlightEnd);
```

Comment:
I'd remove this. You can (safely) add the `mouseup` listener inside `onHighlightStart` & then remove the `highlightEnd` listener from within itself.

```js
onHighlightStart = e => {
  this._highlightState.mouseX = e.clientX;
  this._highlightState.start = true;
  e.target.addEventListener('mouseup', this.onHighlightEnd);
}

onHighlightEnd = e => {
  // ... everything it's already doing
  // at the very end:
  e.target.removeEventListener('mouseup', this.onHighlightEnd);
}
```

You should also be consistent with the event listener names imo (`highlightEnd` vs `onHighlightEnd`).

Reply:
Of course, you can also do `this.base` instead of `e.target`~

Reply:
make sense